.github/workflows/ci.yml

```diff
@@ -3,108 +3,89 @@ on:
   push:
     branches:
       - master
-      - '2.x'
     paths-ignore:
-      - '*.md'
+      - "*.md"
   pull_request:
     paths-ignore:
-      - '*.md'
+      - "*.md"
+permissions:
+  contents: read
 
 jobs:
   test:
+    name: Test - Node.js ${{ matrix.node-version }}
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        name:
-        - Node.js 18.x
-        - Node.js 20.x
-        - Node.js 22.x
-
-        include:
-        - name: Node.js 18.x
-          node-version: "18"
-
-        - name: Node.js 20.x
-          node-version: "20"
-
-        - name: Node.js 22.x
-          node-version: "22"
+        node-version: [18, 20, 22]
 
     steps:
-    - uses: actions/checkout@v4
-
-    - name: Install Node.js ${{ matrix.node-version }}
-      shell: bash -eo pipefail -l {0}
-      run: |
-        nvm install --default ${{ matrix.node-version }}
-        dirname "$(nvm which ${{ matrix.node-version }})" >> "$GITHUB_PATH"
-
-    - name: Configure npm
-      run: |
-        if [[ "$(npm config get package-lock)" == "true" ]]; then
-          npm config set package-lock false
-        else
-          npm config set shrinkwrap false
-        fi
-
-    - name: Install Node.js dependencies
-      run: npm install
-
-    - name: List environment
-      id: list_env
-      shell: bash
-      run: |
-        echo "node@$(node -v)"
-        echo "npm@$(npm -v)"
-        npm -s ls ||:
-        (npm -s ls --depth=0 ||:) | awk -F'[ @]' 'NR>1 && $2 { print $2 "=" $3 }' >> "$GITHUB_OUTPUT"
-
-    - name: Run tests
-      shell: bash
-      run: |
-        npm run test-ci
-        cp coverage/lcov.info "coverage/${{ matrix.name }}.lcov"
-
-    - name: Lint code
-      if: steps.list_env.outputs.eslint != ''
-      run: npm run lint
-
-    - name: Collect code coverage
-      run: |
-        if [[ -d ./coverage ]]; then
-          mv ./coverage "./${{ matrix.name }}"
-          mkdir ./coverage
-          mv "./${{ matrix.name }}" "./coverage/${{ matrix.name }}"
-        fi
-
-    - name: Upload code coverage
-      uses: actions/upload-artifact@v3
-      with:
-        name: coverage
-        path: ./coverage
-        retention-days: 1
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+          check-latest: true
+
+      - name: Configure npm
+        run: |
+          if [[ "$(npm config get package-lock)" == "true" ]]; then
+            npm config set package-lock false
+          else
+            npm config set shrinkwrap false
+          fi
```

Comment:
Is it necessary to take this step in the workflow?

Reply:
In my opinion, this step is not necessary. However, I chose not to refactor the entire workflow because I'm not fully aware of the project's specific requirements.

Reply:
I think that we can remove that piece as now we are using Node@18 as the minimum. This was inherit from the legacy CI pipeline